pages/settings/wallets/lnc.js

```diff
@@ -0,0 +1,119 @@
+import { XXX_DEFAULT_PASSWORD, useLNC } from '@/components/webln/lnc'
+import { lncSchema } from '@/lib/validate'
+import { useRouter } from 'next/router'
+import { useEffect, useRef } from 'react'
+
+export const getServerSideProps = getGetServerSideProps({ authRequired: true })
+
+export default function LNC () {
+  const { provider, enabledProviders, setProvider } = useWebLNConfigurator()
+  const toaster = useToast()
+  const router = useRouter()
+  const lnc = useLNC()
+  const { status, clearConfig, saveConfig, config, name, unlock } = lnc
+  const isDefault = provider?.name === name
+  const unlocking = useRef(false)
+
+  useEffect(() => {
+    if (!unlocking.current && status === Status.Locked) {
+      unlocking.current = true
+      unlock()
+    }
+  }, [status, unlock])
+
+  const defaultPassword = config?.password === XXX_DEFAULT_PASSWORD
+
+  return (
+    <CenterLayout>
+      <h2>Lightning Node Connect for LND</h2>
+      <h6 className='text-muted text-center pb-3'>use Lightning Node Connect for LND payments</h6>
+      <WalletSecurityBanner />
+      <Form
+        initial={{
+          pairingPhrase: config?.pairingPhrase || '',
+          password: (!config?.password || defaultPassword) ? '' : config.password
+        }}
+        schema={lncSchema}
+        onSubmit={async ({ isDefault, ...values }) => {
+          try {
+            await clearConfig()
+            await saveConfig(values)
+            if (isDefault) setProvider(lnc)
+            toaster.success('saved settings')
+            router.push('/settings/wallets')
+          } catch (err) {
+            console.error(err)
+            toaster.danger('failed to attach: ' + err.message || err.toString?.())
+          }
+        }}
+      >
+        <PasswordInput
+          label={
+            <div className='d-flex align-items-center'>pairing phrase
+              <Info label='help'>
+                <Text>
+                  {'We only need permissions for the uri `/lnrpc.Lightning/SendPaymentSync`\n\nCreate a budgeted account with narrow permissions:\n\n```$ litcli accounts create --balance <budget>```\n\n```$ litcli sessions add --type custom --account_id <account_id> --uri /lnrpc.Lightning/SendPaymentSync```'}
+                </Text>
+              </Info>
+            </div>
+          }
+          name='pairingPhrase'
+          initialValue={config?.pairingPhrase}
+          newPass={config?.pairingPhrase === undefined}
+          required
+          autoFocus
+        />
+        <PasswordInput
+          label={<>password <small className='text-muted ms-2'>optional</small></>}
+          name='password'
+          initialValue={defaultPassword ? '' : config?.password}
+          newPass={config?.password === undefined || defaultPassword}
+          hint='encrypts your pairing phrase when stored locally'
+        />
+        <ClientCheckbox
+          disabled={status !== Status.Enabled || isDefault || enabledProviders?.length === 1}
```

Comment:
Shouldn't this be `!configured` instead of `status !== Status.Enabled` like for the other wallets?

Reply:
I'm not sure. The checkbox rules were a bit unclear to me in examples.

Reply:
> I'm not sure

In that case, I am sure that it should be `!configured` since it doesn't make sense to change the checkbox value (to make it a default) if it isn't even configured.

> The checkbox rules were a bit unclear to me in examples.

It tries to handle these three cases to allow changing the default value:

1) if wallet isn't configured, it's not a default and you can't make it a default
2) if it's already a default, you can't remove the default value. unsetting the default only works by making another provider the default. it's weird UX but it solves the issue that some provider has to be the default and I think it would be even weirder UX if making another provider the default is implicit.
3) if it's the only enabled provider, it must stay the default

_mhh, 3) might be included in 2)_